### DISCUSSION TITLE
Suggestion for 2 Camera + 1 Projector model

### DISCUSSION THREAD

**Discussion by goksubaser** [jakobwilm/slstudio#49] [Q&A]
Hello Sir,

We have been building a 2 Camera + 1 Projector Structured Lightning/Triangulation configuration for our University Project. Our first impression was to triangulate 3 times between camera/projector x2 + camera/camera x1 and "3D register" the Point Clouds together to achieve better accuracy. But we are stuck with a problem. Because we only have an offline data set, we cannot calibrate and capture newly calibrated+scanned data.

Our calibration data in hand has the following matrices: 

R(3x3) matrix between camera1/camera2
T(3x1) matrix between camera1/camera2
K(3x3) matrices of both camera1/camera2
k(1x5) matrices of both camera1/camera2

R(3x3) matrix of camera1/projector
R(3x3) matrix of camera2/projector
P(3x4) (projection) matrix of camera1 to itself
P(3x4) (projection) matrix of camera1 to camera2

Q(4x4) matrix of camera2 to camera 1

With those matrices, we could not find a way to triangulate camera/projector couples. Is there a way to calculate Kp and Tp from those matrices without extra calibration needed?
Or is it safe to assume only triangulating between camera1-camera2 using structured lights only to calculate dense correspondence for an accurate enough Point Cloud?

We could not find a similar paper or project of such therefore any suggestions are welcomed.

Thank you for your time and consideration in advance.
Göksu Başer

**Comment by jakobwilm** [comment]
Hi Göksu,

This sounds like a standard CV question. Remember that P = K * [R | t] and as far as I understand, you should be able to derive the needed matrices from what you have. 

Sincerely

**Reply by jakobwilm** [reply]
Regarding triangulation: I would generally recommend to triangulate between the cameras in a 2-camera SL system.

**Reply by goksubaser** [reply]
Thank you sir, that would be enough.
